xdsl/transforms/memref_to_dsd.py

```diff
@@ -42,7 +43,7 @@ class LowerAllocOpPass(RewritePattern):
     def match_and_rewrite(self, op: memref.AllocOp, rewriter: PatternRewriter, /):
         assert isattr(
             memref_type := op.memref.type,
-            MemRefType.constr(element_type=csl.ZerosOpAttrConstr),
+            MemRefType[ZerosOpAttr].constr(element_type=csl.ZerosOpAttrConstr),
```

Comment:
This is a regression from this change. I believe there is no way to convince pyright of the various types involved without this annotation. See https://github.com/microsoft/pyright/issues/7721